master/internal/db/postgres.go

```diff
@@ -1483,6 +1483,35 @@ WHERE id = :id`, setClause(toUpdate)), validation)
 	return nil
 }
 
+// AddModel adds the model to the database and sets its ID.
+func (db *PgDB) AddModel(m *model.Model) error {
```

Comment:
question: are we adding an endpoint for this?

Reply:
I was going to do an add or update endpoint in the next PR. The only complicating factor is that we don't want the name/creation_time of the model to be mutable after creation and the last_update time should be set inside this function rather than by the users machine when they submit the request. I will likely change this function to handle that logic. Would you rather I just make the change in this PR?

Reply:
Nah, we can put this in another PR.